components/Votes/Votes.tsx

```diff
@@ -1,570 +1,29 @@
-            {isDirty() ? (
-              <>
-                <Button
-                  variant="secondary"
-                  label="Reset Changes"
-                  onClick={() => setSelectedVotes({})}
-                />
-                <ButtonSpacer />
-              </>
-            ) : undefined}
-            {!actionStatus.hidden ? (
-              actionStatus.tooltip ? (
-                <Tooltip label={actionStatus.tooltip}>
-                  <div>
-                    <Button
-                      variant="primary"
-                      label={actionStatus.label}
-                      onClick={actionStatus.onClick}
-                      disabled={actionStatus.disabled}
-                    />
-                  </div>
-                </Tooltip>
-              ) : (
-                <Button
-                  variant="primary"
-                  label={actionStatus.label}
-                  onClick={actionStatus.onClick}
-                  disabled={actionStatus.disabled}
-                />
-              )
-            ) : null}
-          </ButtonInnerWrapper>
-        </ButtonOuterWrapper>
-        {pv ? (
-          <>
-            <Divider />
-            {pv}
-          </>
-        ) : null}
-      </>
-    );
-  }
-  function upcomingView() {
-    const pv = pastView();
-    return (
-      <>
-        <Title>Upcoming votes:</Title>
-        <VoteTimeline />
-        <VotesTableWrapper>
-          <VotesList
-            headings={<VotesTableHeadings activityStatus={activityStatus} />}
-            rows={votesToShow.map((vote) => (
-              <VotesListItem
-                vote={vote}
-                phase={phase}
-                selectedVote={undefined}
-                selectVote={() => null}
-                clearVote={() => null}
-                activityStatus={activityStatus}
-                moreDetailsAction={() => openVotePanel(vote)}
-                key={vote.uniqueKey}
-                delegationStatus={getDelegationStatus()}
-                isFetching={
-                  getUserDependentIsFetching() ||
-                  isCommittingVotes ||
-                  isRevealingVotes
-                }
-              />
-            ))}
-          />
-        </VotesTableWrapper>
-        {determineVotesToShow().length > defaultResultsPerPage && (
-          <PaginationWrapper>
-            <Pagination
-              entries={votesToShow}
-              setEntriesToShow={setVotesToShow}
-            />
-          </PaginationWrapper>
-        )}
-        {pv ? (
-          <>
-            <Divider />
-            {pv}
-          </>
-        ) : null}
-      </>
-    );
-  }
-  if (activityStatus === "active") {
-    return activeView();
-  } else if (activityStatus === "upcoming") {
-    return upcomingView();
-  } else {
-    return pastView();
-  }
+  const { activeVotesList, pastVotesList, upcomingVotesList } =
+    useVotesContext();
+
+  const hasActiveVotes = activeVotesList.length > 0;
```

Comment:
This structure actually works out quite conveniently.

Active votes always go on the top. 

If we have active and upcoming votes, the upcoming votes go under the active votes.

If we have upcoming with no active, the upcoming will then be at the top because the active won't show.

Past votes always go to the bottom. If there are no active or upcoming, the past votes will then be the only list shown.